text/0027-manual-disabling-of-flaky-tests.md

```diff
@@ -0,0 +1,75 @@
+* Start Date: 2022-10-13
+* RFC Type: informational
+* RFC PR: https://github.com/getsentry/rfcs/pull/27
+
+# Summary
+
+flaky tests in `sentry` / `getsentry` significantly impact developer
+productivity.  ideally an automated system would improve this situation but
+in the interim this rfc aims to document a manual process for disabling
+flaky tests.
+
+# Motivation
+
+currently any test flake will require the entire job to be rerun, which
+usually wastes about ~20 minutes of developer time per test flake.
+
+to minimize disruptiveness, this rfc proposes manually disabling with a
+procedure.
+
+# Supporting Data
+
+https://sentry.io/organizations/sentry/dashboard/7997/?project=5350637&project=4857230&project=2423079&statsPeriod=14d
+
+# Options Considered
+
+## manual disabling process
+
+**a test should be disabled if it has flaked on the primary branch**
```

Comment:
Why only the primary branch? This is something currently plaguing our PRs in sentry-cocoa, and we recently came to a decision that's close to what this RFC outlines. My current process is that if a test fails and then passes after I retry it in a PR, I immediately open an issue for it and a separate PR to disable the test.

Reply:
yeah there's a bit of nuance to what I've said here -- usually you'll _find_ the flake on a PR and then it's useful to cross reference against the primary branch to make sure it has flaked there at least once

if it's only on the PR then it's probably the PR's fault

admittedly this rfc mostly has sentry / getsentry in mind and there's good instrumentation of the testsuite as sentry issues which "enforces" that this isn't a "my branch only" problem

Reply:
I get what you're saying about stronger verification that a PR didn't cause a failure. I can't see any way to assign cause to either a PR's changes or the test itself other than on a case-by-case basis, that'll be up to authors, reviewers and maintainers.

The perniciousness of a flaky test is the false negative (flaky test passing): you see it fail in a pr, then pass on a retry (indicating it's probably flaky) and then it passes again on the main branch after merging.

Once I see an indication that something is up with a test flipping back and forth, that's cause for further investigation. If a code change causes a failure, that test should _consistently_ fail, otherwise the test is unreliable. 

At that point you can't make any inference on whether it should fail on the main branch since flakiness is by nature indeterminate; it could go on passing for a long time just due to chance.